x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/mapper/PointFieldMapper.java

```diff
@@ -0,0 +1,323 @@
+ *
+ * Uses lucene 8 XYPoint encoding
+ */
+public class PointFieldMapper extends FieldMapper implements ArrayValueMapperParser {
+    public static final String CONTENT_TYPE = "point";
+
+    public static class Names {
+        public static final String IGNORE_MALFORMED = "ignore_malformed";
+        public static final String NULL_VALUE = "null_value";
+    }
+
+    public static class Defaults {
+        public static final Explicit<Boolean> IGNORE_MALFORMED = new Explicit<>(false, false);
+        public static final PointFieldType FIELD_TYPE = new PointFieldType();
+
+        static {
+            FIELD_TYPE.setTokenized(false);
+            FIELD_TYPE.setHasDocValues(true);
+            FIELD_TYPE.setDimensions(2, Integer.BYTES);
+            FIELD_TYPE.freeze();
+        }
+    }
+
+    public static class Builder extends FieldMapper.Builder<Builder, PointFieldMapper> {
+        protected Boolean ignoreMalformed;
+
+        public Builder(String name) {
+            super(name, Defaults.FIELD_TYPE, Defaults.FIELD_TYPE);
+            builder = this;
+        }
+
+        public Builder ignoreMalformed(boolean ignoreMalformed) {
+            this.ignoreMalformed = ignoreMalformed;
+            return builder;
+        }
+
+        protected Explicit<Boolean> ignoreMalformed(BuilderContext context) {
+            if (ignoreMalformed != null) {
+                return new Explicit<>(ignoreMalformed, true);
+            }
+            if (context.indexSettings() != null) {
+                return new Explicit<>(IGNORE_MALFORMED_SETTING.get(context.indexSettings()), false);
+            }
+            return PointFieldMapper.Defaults.IGNORE_MALFORMED;
+        }
+
+        public PointFieldMapper build(BuilderContext context, String simpleName, MappedFieldType fieldType,
+                                      MappedFieldType defaultFieldType, Settings indexSettings,
+                                      MultiFields multiFields, Explicit<Boolean> ignoreMalformed,
+                                      CopyTo copyTo) {
+            setupFieldType(context);
+            return new PointFieldMapper(simpleName, fieldType, defaultFieldType, indexSettings, multiFields,
+                ignoreMalformed, copyTo);
+        }
+
+        @Override
+        public PointFieldType fieldType() {
+            return (PointFieldType)fieldType;
+        }
+
+        @Override
+        public PointFieldMapper build(BuilderContext context) {
+            return build(context, name, fieldType, defaultFieldType, context.indexSettings(),
+                multiFieldsBuilder.build(this, context), ignoreMalformed(context), copyTo);
+        }
+
+        @Override
+        protected void setupFieldType(BuilderContext context) {
+            super.setupFieldType(context);
+
+            fieldType().setGeometryQueryBuilder(new ShapeQueryPointProcessor());
+        }
+    }
+
+    public static class TypeParser implements Mapper.TypeParser {
```

Comment:
I'm good with this as is for this PR, but we will likely want to consider adding `Z` values to points like we do for `GeoPoint`. This way users can add a third dimension to source only. Future PR is fine; at which time we may want to consider sharing code by deriving from `GeoPointFieldMapper.TypeParser`.

Reply:
I agree to keep consistent with Shape field